src/coreComponents/meshUtilities/CPMesh/CPMeshBuilder.hpp

```diff
@@ -0,0 +1,262 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file CPMeshBuilder.hpp
+ */
```

Comment:
When I was working on the LvArray documentation comments such as these were necessary for Doxygen to add the header to some of its lists. I can't say exactly what it gets you, but it was worth it.

Reply:
Yeah, it's what enables Doxygen to generate file synopsis pages like [this one](https://geosx-geosx.readthedocs-hosted.com/en/latest/doxygen_output/html/_data_types_8hpp.html). It's a bit strange that it can't do that by default, just using the file name itself, but that's how it is. We have these `@file` tags in every file of the project.

Reply:
Ok then. And [doc](https://www.doxygen.nl/manual/commands.html#cmdfile) says that maybe just `@file` is enough?

Reply:
> Ok then. And [doc](https://www.doxygen.nl/manual/commands.html#cmdfile) says that maybe just `@file` is enough?

Yes, by the looks of it. What's also interesting is that we include this tag even in `.cpp` files, although nothing in them needs to be documented (for doxygen, anyway). I'm not really sure why we're doing that.